apps/roam/src/utils/hyde.ts

```diff
@@ -530,3 +530,61 @@ export const performHydeSearch = async ({
   }
   return [];
 };
+
+export type VectorMatch = {
+  node: Result;
+  score: number;
+};
+
+export const findSimilarNodesVectorOnly = async ({
+  text,
+  threshold = 0.4,
+  limit = 15,
+}: {
+  text: string;
+  threshold?: number;
+  limit?: number;
+}): Promise<VectorMatch[]> => {
+  if (!text.trim()) {
+    return [];
+  }
+
+  try {
+    const supabase = await getLoggedInClient();
+    if (!supabase) return [];
+
+    const queryEmbedding = await createEmbedding(text);
+
+    const { data, error } = await supabase.rpc("match_content_embeddings", {
+      query_embedding: JSON.stringify(queryEmbedding),
+      match_threshold: threshold,
+      match_count: limit,
+    });
+
+    if (error) {
+      console.error("Vector search failed:", error);
+      throw error;
+    }
+
+    if (!data || !Array.isArray(data)) return [];
+
+    const results: VectorMatch[] = data.map(
```

Comment:
This can be similified to 

```typescript
    const results: VectorMatch[] = data.map((item) => ({
      node: {
        uid: item.roam_uid,
        text: item.text_content,
      },
      score: item.similarity,
    }));
```